File: app/src/main/java/ceui/lisa/adapters/AbstractIllustAdapter.java
package ceui.lisa.adapters;

import android.app.Activity;
import android.content.Context;
import android.content.Intent;

import androidx.annotation.NonNull;
import androidx.core.app.ActivityOptionsCompat;
import androidx.recyclerview.widget.RecyclerView;

import ceui.lisa.R;
import ceui.lisa.activities.ImageDetailActivity;
import ceui.lisa.models.IllustsBean;

public abstract class AbstractIllustAdapter<VH extends RecyclerView.ViewHolder>
        extends RecyclerView.Adapter<VH> {

    protected IllustsBean allIllust;
    protected Context mContext;
    protected int imageSize;
    protected boolean isForceOriginal;
    private long lastClickTime = 0;

    @Override
    public int getItemCount() {
        return allIllust.getPage_count();
    }

    @Override
    public void onBindViewHolder(@NonNull VH holder, int position) {
        holder.itemView.setOnClickListener(v -> {
            if (android.os.SystemClock.elapsedRealtime() - lastClickTime < 1000) {
                return;
            }
            lastClickTime = android.os.SystemClock.elapsedRealtime();

            Intent intent = new Intent(mContext, ImageDetailActivity.class);
            intent.putExtra("illust", allIllust);
            intent.putExtra("dataType", "二级详情");
            intent.putExtra("index", position);
            if (mContext instanceof Activity) {
                ActivityOptionsCompat options = ActivityOptionsCompat.makeSceneTransitionAnimation(
                    (Activity) mContext,
                    holder.itemView.findViewById(R.id.illust_image),
                    "image_" + position
                );
                mContext.startActivity(intent, options.toBundle());
            } else {
                mContext.startActivity(intent);
            }
        });
    }
}

File: app/src/main/java/ceui/lisa/adapters/AbstractIllustAdapter.kt
package ceui.lisa.adapters

import android.content.Context
import android.view.LayoutInflater
import android.view.ViewGroup
import android.os.SystemClock
import androidx.recyclerview.widget.RecyclerView
import ceui.pixiv.route.AppRoute
import ceui.lisa.models.IllustsBean

abstract class AbstractIllustAdapter<VH : RecyclerView.ViewHolder> : RecyclerView.Adapter<VH>() {

    @JvmField
    protected var allIllust: IllustsBean? = null

    @JvmField
    protected var mContext: Context? = null

    @JvmField
    protected var imageSize: Int = 0

    @JvmField
    protected var isForceOriginal: Boolean = false

    private var lastClickTime: Long = 0

    override fun getItemCount(): Int {
        return allIllust?.page_count ?: 0
    }

    override fun onBindViewHolder(holder: VH, position: Int) {
        holder.itemView.setOnClickListener {
            if (SystemClock.elapsedRealtime() - lastClickTime < 1000) {
                return@setOnClickListener
            }
            lastClickTime = SystemClock.elapsedRealtime()
            AppRoute.ImageDetail(allIllust, position).start(mContext!!)
        }
    }
}